### DISCUSSION TITLE
ArrayItems如何做到动态schema

### DISCUSSION THREAD

**Discussion by xyyit9** [alibaba/formily#2828] [Q&A 2.x]
codesandbox:  https://codesandbox.io/s/modest-resonance-61b57?file=/App.tsx:3310-3320
预期：选择【选项一】，出现input输入框，选择【选项二】，出现select选择框。在只有一个ArrayItem的时候，符合预期。但是当添加了多个ArrayItem的时候，并不符合预期，所有的ArrayItem，都会出现和【选项一】一样的内容。
有两个问题：
1. 如何做到不同ArrayItems会根据自己当前的选项，动态展示？
2.当前只有第一个ArrayItem可以控制表单，其他ArrayItem并不能控制，如何做到其他ArrayItem也能控制呢？ 

预期是这样：
![image](https://user-images.githubusercontent.com/31262841/154190860-5b24a5e9-1f84-4361-91fd-e8bda6216cfe.png)

**Comment by wittech** [comment]
自己来扩展自定义组件灵活点

**Comment by wittech** [comment]
@xyyit9 https://formilyjs.org/zh-CN/guide/advanced/controlled 参考这个
